kubeflow/trainer/backends/kubernetes/backend.py

```diff
@@ -410,6 +410,105 @@ def delete_job(self, name: str):
 
         logger.debug(f"{constants.TRAINJOB_KIND} {self.namespace}/{name} has been deleted")
 
+    def get_job_events(self, name: str) -> list[types.Event]:
```

Comment:
Please implement similar API in the OptimizerClient too: https://github.com/kubeflow/sdk/blob/9db46db71968c6ab9bdd51b726c01a4344ea87c1/kubeflow/optimizer/backends/kubernetes/backend.py